packages/ses/src/immutable-array-buffer-shim.js

```diff
@@ -9,88 +8,99 @@ import {
   TypeError,
 } from './commons.js';
 
-/**
- * This class only exists within the shim, as a convience for imperfectly
- * emulating the proposal, which would not have this class. In the proposal,
- * `transferToImmutable` makes a new `ArrayBuffer` that inherits from
- * `ArrayBuffer.prototype` as you'd expect. In the shim, `transferToImmutable`
- * makes a normal object that inherits from
- * `ImmutableArrayBufferInternal.prototype`, which has been surgically
- * altered to inherit from `ArrayBuffer.prototype`. The constructor is
- * captured for use internal to this module, and is made otherwise inaccessible.
- * Therefore, `ImmutableArrayBufferInternal.prototype` and all its methods
- * and accessor functions effectively become hidden intrinsics.
- *
- * TODO handle them as hidden intrinsics, so they get hardened when they should.
- */
-class ImmutableArrayBufferInternal {
-  /** @type {ArrayBuffer} */
-  #buffer;
+// If it already exists, don't replace it with the shim.
+if (!('transferToImmutable' in arrayBufferPrototype)) {
```

Comment:
Conditional shims for non stage 4 features have been seen as problematic by the community as there is a risk that the program would start relying on a shim behavior that does not match what the final spec says, resulting in breakage when the engine starts implementing the feature.

Reply:
Makes sense. Will do. Is there something I can cite?

Reply:
I believe there is an article out there but I couldn't find a link

Reply:
I simply cited your comment above. Done.

Reply:
> Conditional shims for non stage 4 features have been seen as problematic by the community as there is a risk that the program would start relying on a shim behavior that does not match what the final spec says, resulting in breakage when the engine starts implementing the feature... I believe there is an article out there but I couldn't find a link

https://ponyfill.com (which conveniently also links to http://svdictionary.com/words/ponyfill and https://ponyfoo.com/articles/polyfills-or-ponyfills and https://kikobeats.com/polyfill-ponyfill-and-prollyfill/ ).

Summary: for work like this, we can export a function for use like `transferToImmutable(arrayBuffer)` and `transferToImmutable(arrayBuffer, newLength)`, but SHOULD NOT add any new methods or properties to ArrayBuffer.prototype.